datadog/resource_datadog_role.go

```diff
@@ -228,14 +250,15 @@ func updateRolePermissionsState(ctx context.Context, d *schema.ResourceData, rol
 	return nil
 }
 
-func appendPerm(perms []map[string]string, permID string, permsIDToName map[string]string) []map[string]string {
-	// If perm ID is not restricted, add it to the state
-	if permName, ok := permsIDToName[permID]; ok {
+func appendPerm(perms []map[string]string, permID string, permIDToAttributes map[string]PermAttributes, isOptedOutOfDefaultPermissions bool) []map[string]string {
+	if permAttributes, ok := permIDToAttributes[permID]; ok {
 		permR := map[string]string{
 			"id":   permID,
-			"name": permName,
+			"name": permAttributes.Name,
+		}
+		if isOptedOutOfDefaultPermissions || !permAttributes.IsRestricted {
+			perms = append(perms, permR)
 		}
```

Comment:
just want to make sure i understand this part. if it's (a) _not_ opted out of default permissions (i.e., the role will include all restricted permissions), and (b) the permission is restricted, then that permission will _not_ be added to the request. that's because the backend will automatically add _all_ restricted permissions, so they shouldn't be explicitly provided. did i get that right? might be worth a code comment bc it's a little counterintuitive

Reply:
yes your understanding is correct, i'll add code comment to make it more clear